internal/pkg/scaffold/helm/role.go

```diff
@@ -0,0 +1,241 @@
+// Copyright 2019 The Operator-SDK Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package helm
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"sort"
+	"strings"
+
+	"github.com/operator-framework/operator-sdk/internal/pkg/scaffold"
+
+	"github.com/ghodss/yaml"
+	log "github.com/sirupsen/logrus"
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/version"
+	"k8s.io/client-go/discovery"
+	"k8s.io/client-go/rest"
+	"k8s.io/helm/pkg/chartutil"
+	"k8s.io/helm/pkg/manifest"
+	"k8s.io/helm/pkg/proto/hapi/chart"
+	"k8s.io/helm/pkg/renderutil"
+	"k8s.io/helm/pkg/tiller"
+)
+
+// CreateRoleScaffold generates a role scaffold from the provided helm chart. It
+// renders a release manifest using the chart's default values and uses the Kubernetes
+// discovery API to lookup each resource in the resulting manifest.
+func CreateRoleScaffold(cfg *rest.Config, chart *chart.Chart, isClusterScoped bool) (*scaffold.Role, error) {
+	log.Info("Generating RBAC rules")
+
+	roleScaffold := &scaffold.Role{
+		IsClusterScoped:  isClusterScoped,
+		SkipDefaultRules: true,
+		// TODO: enable metrics in helm operator
```

Comment:
What were you thinking?

Reply:
Just to enable the CR metrics in [`helm.Run`](https://github.com/operator-framework/operator-sdk/blob/master/pkg/helm/run.go). E.g.:
* https://github.com/operator-framework/operator-sdk/blob/4011c4da21119035d61b993abdad6987acbbe71b/internal/pkg/scaffold/cmd.go#L125
* https://github.com/operator-framework/operator-sdk/blob/4011c4da21119035d61b993abdad6987acbbe71b/internal/pkg/scaffold/cmd.go#L146-L150